corehq/apps/fixtures/fixturegenerators.py

```diff
@@ -45,42 +48,81 @@ class ItemListsProvider(FixtureProvider):
 
     def __call__(self, restore_state):
         restore_user = restore_state.restore_user
-
-        all_types = {t._id: t for t in FixtureDataType.by_domain(restore_user.domain)}
-        global_types = {id: t for id, t in all_types.items() if t.is_global}
-
+        global_types = {}
+        user_types = {}
+        for data_type in FixtureDataType.by_domain(restore_user.domain):
+            if data_type.is_global:
+                global_types[data_type._id] = data_type
+            else:
+                user_types[data_type._id] = data_type
+        items = self.get_global_items(global_types, restore_state)
+        items.extend(self.get_user_items(user_types, restore_user))
+        return items
+
+    def get_global_items(self, global_types, restore_state):
+        restore_user = restore_state.restore_user
+        domain = restore_user.domain
+        db = get_blob_db()
+        if not restore_state.overwrite_cache:
+            try:
+                fixture_data = db.get(domain, FIXTURE_BUCKET).read()
+                return [fixture_data] if fixture_data else []
+            except NotFound:
+                pass
+        global_items = self._get_global_items(global_types, restore_user)
```

Comment:
this will cache, globally for the domain, XML that includes this `restore_user.user_id`. Looks like that issue was reported in [FB257817](https://manage.dimagi.com/default.asp?257817), but not fixed by #16951. It's unclear to me if this is a blocker or not, but I'm guessing it is.

Reply:
oh man, I guess this explains why some fixtures were just not showing up for users - they were assigned to the user who initially created the cached restore.